src/chunk_dispatch_state.c

```diff
@@ -84,6 +84,7 @@ chunk_dispatch_exec(CustomScanState *node)
 
 		/* Find or create the insert state matching the point */
 		cis = ts_chunk_dispatch_get_chunk_insert_state(dispatch, point);
+		ts_chunk_insert_state_switch(cis, dispatch);
```

Comment:
Instead of calling ts_chunk_insert_state_switch , can ts_chunk_dispatch_get_chunk_insert_state return cis with adjusted slots when it is a cached  CIS. Dispatch shouldn't have to take care of this. It is internal to CIS state.

Reply:
Sorry I am not sure I follow, you rather have `ts_chunk_insert_state_switch(cis, dispatch)`; be called from `ts_chunk_dispatch_get_chunk_insert_state` Because something needs to call the switch functions since the slot is shared across CISs.